ganttproject-tester/test/biz/ganttproject/impex/csv/CsvImportTest.java

```diff
@@ -149,115 +92,163 @@ public void testSkipLinesWithEmptyMandatoryFields() throws Exception {
     String data2 = "a2,b2,c2";
     String data3 = ",b3,c3";
 
-    AtomicBoolean wasCalled = new AtomicBoolean(false);
-    RecordGroup recordGroup = createSkipLinesWithEmptyMandatoryFieldsRecordGroup(wasCalled);
-    GanttCSVOpen importer = new GanttCSVOpen(createSupplier(Joiner.on('\n').join(header, data1, data2, data3)), CSV, recordGroup);
-    importer.load();
-    assertTrue(wasCalled.get());
-    assertEquals(2, importer.getSkippedLineCount());
-
-    wasCalled = new AtomicBoolean(false);
-    recordGroup = createSkipLinesWithEmptyMandatoryFieldsRecordGroup(wasCalled);
-    importer = new GanttCSVOpen(createSupplier(createXls(header, data1, data2, data3)), XLS, recordGroup);
-    importer.load();
-    assertTrue(wasCalled.get());
-    assertEquals(2, importer.getSkippedLineCount());
+    doTestSkipLinesWithEmptyMandatoryFields(header, data1, data2, data3);
   }
 
-  private RecordGroup createBasicRecordGroup(final AtomicBoolean wasCalled) {
-    return new RecordGroup("AB", ImmutableSet.of("A", "B")) {
-      @Override
-      protected boolean doProcess(SpreadsheetRecord record) {
-        if (!super.doProcess(record)) {
-          return false;
+  private void doTestBasic(String... data) throws Exception {
```

Comment:
you never call it with the number of arguments other than two (in fact, there is just a single call). Make it simpler: replace vararg with header and data and maybe just move the body of this method back into testBasic. 

The same applies to all similar refactorings.